edward/inferences/renyi_divergence.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import six
+import numpy as np
+import tensorflow as tf
+
+from edward.inferences.variational_inference import VariationalInference
+from edward.models import RandomVariable
+from edward.util import copy
+
+try:
+    from edward.models import Normal
+    from tensorflow.contrib.distributions import kl_divergence
+except Exception as e:
+    raise ImportError(
+        "{0}. Your TensorFlow version is not supported.".format(e))
+
+
+class Renyi_divergence(VariationalInference):
+    """Variational inference with the Renyi divergence
+
+    $ \text{D}_{R}^{(\alpha)}(q(z)||p(z \mid x))
+        = \frac{1}{\alpha-1} \log \int q(z)^{\alpha} p(z \mid x)^{1-\alpha} dz $
+
+    To perform the optimization, this class uses the techniques from
+    Renyi Divergence Variational Inference (Y. Li & al, 2016)
+
+    # Notes:
+        - Renyi divergence does not have any analytic version.
+        - Renyi divergence does not have any version for non reparametrizable
+            models.
+        - backward_pass = 'max': (extreme case $\alpha \rightarrow -\infty$)
+            the algorithm chooses the sample that has the maximum unnormalised
+            importance weight. This does not minimize the Renyi divergence
+            anymore.
+        - backward_pass = 'min': (extreme case $\alpha \rightarrow +\infty$)
+            the algorithm chooses the sample that has the minimum unnormalised
+            importance weight. This does not minimize the Renyi divergence
+            anymore. This mode is not describe in the paper but implemented
+            in the publicly available implementation of the paper's experiments.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(Renyi_divergence, self).__init__(*args, **kwargs)
+
+    def initialize(self,
+                   n_samples=32,
+                   alpha=1.,
+                   backward_pass='full',
+                   *args, **kwargs):
+        """Initialize inference algorithm. It initializes hyperparameters
+        and builds ops for the algorithm's computation graph.
+
+        Args:
+            n_samples: int, optional.
+                Number of samples from variational model for calculating
+                stochastic gradients.
+            alpha: float, optional.
+                Renyi divergence coefficient.
```

Comment:
Could be useful to specify the domain of the coefficient. E.g., `Must be greater than 0.` or etc.